src/main/java/com/google/api/generator/gapic/composer/Composer.java

```diff
@@ -210,17 +209,48 @@ static List<GapicClass> prepareExecutableSamples(List<GapicClass> clazzes, Strin
               List<Sample> samples =
                   gapicClass.samples().stream()
                       .map(
-                          sample -> addRegionTagAndHeaderToSample(sample, apiShortName, apiVersion))
+                          sample ->
+                              addRegionTagAndHeaderToSample(
+                                  sample, parseDefaultHost(gapicClass.defaultHost()), apiVersion))
                       .collect(Collectors.toList());
               return gapicClass.withSamples(samples);
             })
         .collect(Collectors.toList());
   }
 
-  private static Sample addRegionTagAndHeaderToSample(
+  // Parse defaultHost for apiShortName. Need to account for regional default endpoints like
+  // "us-east1-pubsub.googleapis.com".
+  // Will keep backup name parsed from proto package for now.
+  @VisibleForTesting
+  protected static String parseDefaultHost(String defaultHost) {
+    String apiShortName;
+    String apiShortNameHelper;
+    // If the defaultHost is of the format "**.googleapis.com", take the name before the first
+    // period.
+    Pattern standardApiHost = Pattern.compile("\\.");
+    // If the defaultHost is of the format "**-**-**.googleapis.com", take the section before the
+    // first period and after the last dash to follow CSharp's implementation here:
+    // https://github.com/googleapis/gapic-generator-csharp/blob/main/Google.Api.Generator/Generation/ServiceDetails.cs#L70
+    Pattern regionalApiHost = Pattern.compile("-");
+    apiShortNameHelper = standardApiHost.split(defaultHost)[0];
```

Comment:
Try to use `StringSplitter` from Guava instead of the default String split, see https://errorprone.info/bugpattern/StringSplitter

Reply:
Thanks for the tip! Makes sense; updated.